realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -996,6 +998,19 @@ public void where_findAll_size() {
     }
 
     @Test
+    public void size_returns_Integer_MAX_VALUE_for_huge_results() {
```

Comment:
This test is highly dependant on the implementation of RealmResults, but there is probably no way around that.